packages/headless/src/modules/customer-auth/services/index.ts

```diff
@@ -66,4 +72,28 @@ export class CustomerAuthService {
     const token = this.jwtService.sign(payload);
     return this.helper.serviceResponse.successResponse({ token }, HttpStatus.OK,);
   }
+
+  async forgotPassword(data: CustomerForgotPasswordRequest, baseUrl: string): Promise<CustomerForgotPasswordResponse> {
+
+    const doesCustomerEmailExist = data.email && await this.customerRepo.getCustomerPassword({ email: data.email });
+    const doesCustomerPhoneExist = data.phone && await this.customerRepo.getCustomerPassword({ phone: data.phone });
+    if (!doesCustomerEmailExist && !doesCustomerPhoneExist) return this.helper.serviceResponse.errorResponse(CustomerForgotPasswordErrorMessages.CAN_NOT_GET_CUSTOMER, null, HttpStatus.BAD_REQUEST,);
+
+    const customer = doesCustomerEmailExist || doesCustomerPhoneExist;
+
+    customer.resetPasswordToken = token;
+    customer.resetPasswordExpires = Date.now() + ONE_HOUR;
+
+    const updatedCustomer = await this.customerRepo.updateCustomer(customer.id, customer);
+    if (!updatedCustomer) return this.helper.serviceResponse.errorResponse(CustomerForgotPasswordErrorMessages.CAN_NOT_UPDATE_CUSTOMER_PASSWORD, null, HttpStatus.BAD_REQUEST);
+
+    const resetUrl = baseUrl + process.env.AUTH_RESET_ORIGINAL_URL || 'customer/auth/reset/' + token;
```

Comment:
in this url, there should be a handler. So you'll have to create a dynamic route. That's complicated. I think you can keep a fixed url with a dynamic token appended with that url.

Reply:
Also there's another problem. you are allowing the customer to reset his token, but if a customer does not have a mail then how he will be able to reset his password?
let's discuss it in the next day.

Reply:
If the customer does not have an email address, we will send this URL to the customer's phone number via SMS. because this code is executed for a customer's email or phone number. But now we do not have any phone SMS processes. We will discuss the phone SMS API.